File: hacking_with_python/ssrf.py
import requests

def test_ssrf_vulnerability(url):
    try:
        response = requests.get(url)
        if response.status_code == 200:
            print("SSRF Vulnerability Found: Accessible URL -", url)
        else:
            print("URL is not accessible or SSRF Vulnerability not present.")
    except Exception as e:
        print("Error occurred:", str(e))

if __name__ == "__main__":
    target_website = input("Enter your target website link: ")
    test_ssrf_vulnerability(target_website)

File: hacking_with_ruby/ssrf.rb
require 'net/http'

def test_ssrf_vulnerability(url)
  uri = URI.parse(url)

  # Check if the URI scheme is allowed
  if ['http', 'https'].include?(uri.scheme)
    response = Net::HTTP.get_response(uri)
    
    # Check if the response status code indicates a successful request
    if response.code.to_i == 200
      puts "Vulnerability found: #{url} is vulnerable to SSRF."
    else
      puts "#{url} is not vulnerable to SSRF."
    end
  else
    puts "Invalid URI scheme: #{uri.scheme}. SSRF may not be possible."
  end
rescue URI::InvalidURIError => e
  puts "Invalid URL: #{url}. #{e.message}"
rescue SocketError => e
  puts "Failed to connect to #{url}. #{e.message}"
rescue => e
  puts "An error occurred: #{e.message}"
end

puts "Enter your target website url: "
target_url = gets.chomp

# Example usage:
test_ssrf_vulnerability(target_url)
